File: Fibonacci/TypeScript/fibonacci.ts
function fibonacci(length: number) {
  const result: Array<number> = [1, 1];

  // Base array is the first two digits of the sequence.
  // For loop starts from index 2, adding the previous
  // two values of result, and adding that to the array.
  // This sequence progresses until i is equal to length.
  for (let i = 2; i < length; i++) {
    result[i] = result[i - 1] + result[i - 2];
  }

  return result;
}

console.log(fibonacci(5));

File: Go/Fibonacci/fibonacci.go
package main

import "fmt"

func fibonacci(ln int) (sequence []int) {
    if ln <= 0 {
        sequence = []int{0}
        return 
    }

    if ln == 1 {
        sequence = []int{1}
        return 
    } 

    sequence = make([]int, ln)
    sequence[0], sequence[1] = 1, 1

    for i := 2; i < ln; i++ {
        sequence[i] = sequence[i - 1] + sequence[i - 2]
    }

    return 
}

func main() {
    fmt.Println(fibonacci(25))
}

